### PR TITLE
Features/khmic/elevation

### PR DESCRIPTION
**Pull request by khmic5**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java [hunk 1]
```diff
@@ -111,9 +175,8 @@
      * @return the response from a successful Get Data for Bounding Box API.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ElevationResult>> postDataForPointsWithResponse(
-            JsonFormat format, List<LatLongPairAbbreviated> points) {
-        return this.serviceClient.postDataForPointsWithResponseAsync(format, points);
+    public Mono<Response<ElevationResult>> postDataForPointsWithResponse(List<GeoPosition> points) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java]: Here the pattern has been: 

1. Expose a single Public API (e.g. getDataForPoints())
2. Inside this API, make a decision: if the number of points is smaller than a certain threshold, then use the GET api, else use the POST api. 

This is valid for DataForPoints and Polyline. Let's consolidate methods here.

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java [hunk 2]
```diff
@@ -133,14 +196,15 @@
...
+     * @param context The context to associate with this operation.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorResponseException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return the response from a successful Get Data for Bounding Box API.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ElevationResult> postDataForPoints(JsonFormat format, List<LatLongPairAbbreviated> points) {
-        return this.serviceClient.postDataForPointsAsync(format, points);
+    Mono<Response<ElevationResult>> postDataForPointsWithResponse(List<GeoPosition> points, Context context) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java]: same as above

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java [hunk 3]
```diff
@@ -174,9 +237,45 @@
      * @return the response from a successful Get Data for Bounding Box API.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ElevationResult>> getDataForPolylineWithResponse(
-            JsonFormat format, List<String> lines, Integer samples) {
-        return this.serviceClient.getDataForPolylineWithResponseAsync(format, lines, samples);
+    public Mono<ElevationResult> getDataForPolyline(List<GeoPosition> lines, Integer samples) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java:161]: same as above

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java [hunk 4]
```diff
@@ -245,9 +344,45 @@
      * @return the response from a successful Get Data for Bounding Box API.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ElevationResult>> postDataForPolylineWithResponse(
-            JsonFormat format, List<LatLongPairAbbreviated> polyline, Integer samples) {
-        return this.serviceClient.postDataForPolylineWithResponseAsync(format, polyline, samples);
+    public Mono<ElevationResult> postDataForPolyline(List<GeoPosition> polyline, Integer samples) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationAsyncClient.java]: same as above

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationClient.java [hunk 5]
```diff
@@ -144,15 +266,45 @@ public ElevationResult getDataForPolyline(JsonFormat format, List<String> lines,
...
      *     should be provided along a polyline path. The number of samples should range from 2 to 2,000. Default value
      *     is 10.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response from a successful Get Data for Bounding Box API.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<ElevationResult> postDataForPolylineWithResponse(List<GeoPosition> polyline, Integer samples, Context context) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationClient.java]: same as teh async client - 1 API only

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.List;
+
+import com.azure.core.models.GeoBoundingBox;
+import com.azure.core.models.GeoPosition;
+import com.azure.maps.elevation.implementation.models.LatLongPairAbbreviated;
+
+public class Utility {
+    
+    public static List<LatLongPairAbbreviated> toLatLongPairAbbreviated(List<GeoPosition> points) {
+        List<LatLongPairAbbreviated> points2 = new ArrayList<>();
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java]: "points2"? ⛔

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        for (GeoPosition point : points) {
+            points2.add(new LatLongPairAbbreviated().setLat(point.getLatitude()).setLon(point.getLongitude()));
+        }
+        return points2;
+    }
+
+    public static List<String> geoPositionToString(List<GeoPosition> points) {
+        List<String> points2 = new ArrayList<>();
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java]: same

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        for (GeoPosition point : points) {
+            points2.add(point.getLongitude() + "," + point.getLatitude());
+        }
+        return points2;
+    }
+
+    public static List<Float> geoBoundingBoxToFloat(GeoBoundingBox boundingBox) {
```
Comment (daniel-rocha) [inline sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java]: you may want to re-check this in the swagger. coordinates should be represented as double, not float. it's not a terribly big deal but we need to be consistent.

**Review by daniel-rocha — CHANGES_REQUESTED** [review]

File: sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java (hunk 8 above)
Comment (khmic5) [reply sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/implementation/helpers/Utility.java]: It is currently format: float in swagger, should I introduce and push changes in swagger for 1. changing float to double and 2. deleting the 2 lines of "x-ms-skip-url-encoding": true?

**Review by daniel-rocha — APPROVED** [review]

**Pull request merged**